plugins/events-backend-module-github/src/http/createGithubSignatureValidator.ts

```diff
@@ -36,10 +37,25 @@ import { verify } from '@octokit/webhooks-methods';
 export function createGithubSignatureValidator(
   config: Config,
 ): RequestValidator | undefined {
-  const secret = config.getOptionalString(
+  const webhookSecrets = new Set<string>();
+
+  const integrations = ScmIntegrations.fromConfig(config);
+  for (const integration of integrations.github.list()) {
```

Comment:
do we have info that allows us to select the correct integration for each event? would be a bit more efficient and safer

Reply:
Turns out that it's a bit complex.

The config contains app IDs, but the only thing you can get out of webhooks are app _installation_ IDs. So I have to do the octokit dance to make requests to resolve installation IDs to app IDs to find the right secret.

Reply:
OK, added a full implementation for that, let me know what you think. It came at a pretty high cost but, once we make something like an octokit provider into a core integrations thing, this might be less of an issue.